src/MLJGLMInterface.jl

```diff
@@ -87,17 +95,19 @@ Augment the matrix `X` with a column of ones if the intercept is to be
 fitted (`b=true`), return `X` otherwise.
 """
 function augment_X(X::Matrix, b::Bool)::Matrix
-    b && return hcat(X, ones(eltype(X), size(X, 1), 1))
+    b && return hcat(X, ones(float(Int), size(X, 1), 1))
     return X
 end
 
 _to_vector(v::Vector) = v
 _to_vector(v) = collect(v)
+_to_array(v::AbstractArray) = v
+_to_array(v) = collect(v)
 
 """
     split_X_offset(X, offsetcol::Nothing)
 
-When no offset is specied, return X and an empty vector.
+When no offset is specied, return X, an empty vector.
```

Comment:
```suggestion
When no offset is specified, return `X` and an empty vector.
```
The method still returns `(X, Float64[])` right?